File: leetcode/217-contains-duplicate/main.go
package main

func containsDuplicate(nums []int) bool {
	hash := make(map[int]bool)
	for i := 0; i < len(nums); i++ {
		if hash[nums[i]] {
			return true
		}
		hash[nums[i]] = true
	}
	return false
}

func main() {

}

File: leetcode/217-contains-duplicate/main.py
"""
    1st approach: hashtable

    Time    O(n)
    Space   O(n)
    100 ms, faster than 90.95%
"""


class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        hs = set()
        for num in nums:
            if num in hs:
                return True
            hs.add(num)
        return False
